### PR TITLE
header parser added to http2 workflow (#755)

### PR DESCRIPTION
**Pull request by gosubpl**
Refs #755

### CODE REVIEW

**Comment by gosubpl** [comment]
Supersedes https://github.com/akka/akka-http/pull/800

**Comment by gosubpl** [comment]
Correctness checks moved to https://github.com/akka/akka-http/issues/812

**Comment by gosubpl** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
I pushed into your branch removing the file mode changes. Implementation LGTM. Could you implement the pending test in `Http2ServerSpec.scala` with a small example?

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala [hunk 1]
```diff
@@ -66,8 +67,12 @@ private[http2] object RequestParsing {
           rec(remainingHeaders.tail, method, scheme, authority, path, contentType, contentLength, headers)
 
         case (name, value) ⇒
-          // FIXME: parse to real headers
-          rec(remainingHeaders.tail, method, scheme, authority, path, contentType, contentLength, headers += RawHeader(name, value))
+          // FIXME: later modify by adding HttpHeaderParser.parseHttp2Header that would use (name, value) pair directly
+          //        or use a separate, simpler, parser for Http2
+          val concHeaderLine = name + ": " + value + "\r\nx"
```
Comment (ktoso) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala]: is the x on purpose?

**Comment by ktoso** [comment]
LGTM

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala (hunk 1 above)
Comment (jrudolph) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala]: Was explained here: https://github.com/akka/akka-http/pull/800#discussion_r97071257

@gosubpl could you also add a comment into the code for now?

Comment (gosubpl) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala]: yes, will do

**Comment by gosubpl** [comment]
Test with three various headers and a small comment on the origin of 'x' added.
@jrudolph @ktoso thanks for insightful comments.

**Comment by jrudolph** [comment]
Great, thanks @gosubpl.

**Pull request merged**